### ISSUE TITLE
Allow spontaneous interruption without loss of training steps

### ISSUE THREAD

**Issue by nelson-lojo** [in-context-learning-2024/in-context#21]
If we catch [`KeyboardInterrupt` Exceptions](https://docs.python.org/3.10/library/exceptions.html#KeyboardInterrupt) in the training loop, we can save the model outside of yaml-specified checkpoints in the case where checkpointing isn't as fine grained (e.g. in disk-constrained/bottlenecked systems)
